src/nvim/ex_cmds.c

```diff
@@ -1408,32 +1408,30 @@ char_u *make_filter_cmd(char_u *cmd, char_u *itmp, char_u *otmp)
   }
 
   if (itmp != NULL) {
-    strncat(buf, " < ", len);
-    strncat(buf, (char *) itmp, len);
+    strncat(buf, " < ", len - 1);
+    strncat(buf, (const char *)itmp, len - 1);
```

Comment:
@jamessan Buffer is already filled by `vim_snprintf`/`strncpy`, so to actually supply correct length one needs to write different code with more complex calculations. Or do not bother at all, memory should’ve been allocated correctly:

```C
#define STRAPPEND(p, s, len) \
    do {
      const size_t len_ = (len);
      memcpy(p, s, len_);
      p += len;
    } while (0)
char *p = buf;
if (itmp || otmp) {
  const char *const fmt = is_fish_shell ? "begin; %s; end" : "(%s)";
  p += vim_snprintf(buf, len, fmt, (const char *)cmd);
} else {
  STRAPPEND(p, cmd, len);
}
if (itmp) {
  STRAPPEND(p, S_LEN(" < "));
  STRAPPEND(p, itmp, STRLEN(itmp));
}
```